pallets/deeper-node/src/lib.rs

```diff
@@ -77,79 +95,427 @@ decl_module! {
         #[weight = 10_000]
         pub fn unregister_device(origin) -> DispatchResult {
             let sender = ensure_signed(origin)?;
-            ensure!(<DeviceInfo<T>>::contains_key(sender.clone()), "device not registered!");
-            let node = <DeviceInfo<T>>::get(sender.clone());
-            let country = node.country;
-            let _ = Self::try_remove_server(&sender, country);
+            ensure!(<DeviceInfo<T>>::contains_key(&sender), "device not registered!");
+            let _ = Self::try_remove_server(&sender);
             <DeviceInfo<T>>::remove(sender.clone());
             T::Currency::unreserve(&sender,BalanceOf::<T>::from(MIN_LOCK_AMT));
             Self::deposit_event(RawEvent::UnregisterNode(sender));
             Ok(())
         }
 
         #[weight = 10_000]
-        pub fn register_server(origin, country: u16) -> DispatchResult {
+        pub fn register_server(origin, duration: T::BlockNumber) -> DispatchResult {
             let sender = ensure_signed(origin)?;
-            ensure!(country < 1024, "invalid country code");
-            ensure!(<DeviceInfo<T>>::contains_key(sender.clone()),
-                "sender device needs register first");
-
-            let _ = Self::try_add_server(&sender, country);
-
+            ensure!(<DeviceInfo<T>>::contains_key(&sender),
+                    "sender device needs register first");
+            let _ = Self::try_remove_server(&sender);
```

Comment:
Do we need to call try_remove here?

Reply:
I see. the reason is to update duration. maybe it's better we add another call: update_server(origin, duration, ip). I feel that the most frequent call would be update_server by IP and country. Let's discuss it offline.